common/dbus/platform/asio_libdbus/dbus_client.cpp

```diff
@@ -135,16 +142,34 @@ error::Error DBusClient::CallMethod(
 	// We need to create a copy here because we need to make sure the handler,
 	// which might be a lambda, even with captures, will live long enough for
 	// the finished pending call to use it.
-	DBusCallReplyHandler *handler_copy = new DBusCallReplyHandler(handler);
-	if (!dbus_pending_call_set_notify(pending, HandleReply, handler_copy, FreeHandlerCopy)) {
+	auto *handler_copy = new DBusCallReplyHandler<ReplyType>(handler);
+	if (!dbus_pending_call_set_notify(
+			pending, HandleReply<ReplyType>, handler_copy, FreeHandlerCopy<ReplyType>)) {
```

Comment:
Preexisting, but do we leak `handler_copy` here if this call fails?

Even though the lifetime is short, I would consider a `unique_ptr` for `handler_copy`. If we forget to free a raw pointer, it's an invisible bug (yes, asan can catch it, but only if this block hits). OTOH, if we forget to call `unique_ptr::release()` it's a very obvious bug, which will immediately be visible.

Reply:
Yes, you're right. Actually, in this particular case, `dbus_pending_call_set_notify()` only fails if there's not enough memory for it, but that's not a good enough excuse. :)